### PR TITLE
Feat(IP Filtering): Add test cases

### PR DESCRIPTION
**Pull request by yasserfaraazkhan**
Jira: https://mattermost.atlassian.net/browse/MM-55486
Doc:  https://docs.google.com/document/d/1bM88G_sXk1MeKYR-rpX4NFP195bHJq1ttGvWLjDklz4/edit?pli=1

### CODE REVIEW

File: data/test-by-folder.json [hunk 1]
```diff
@@ -2077,6 +2077,15 @@
     "folder": "cloud/in-product-renewal",
     "tests": ["Renewal | Should be available to Renew within the product"]
   },
+  {
+    "folder": "cloud/ip-filtering",
+    "tests": [
+      "IP Filtering feature is available for Enterprise license and Enterprise trial license",
```
Comment (nickmisasi) [inline data/test-by-folder.json:2083]: Nit: For now, this is cloud only

**Review by nickmisasi — APPROVED** [review]
LGTM! Thanks Yasser!

File: data/test-cases/cloud/ip-filtering/test-1.md [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+Test prerequisites:
+
+- Create a workspace and leave it in Enterprise Trial license from https://portal.test.cloud.mattermost.com/
+
+Scenario 1: The Feature flag is Disabled and instance is on Enterprise trial license.
+
+1. In Split.io disable `CloudIPFiltering` feature flag for all test environments.
+2. Login with admin credentials.
+3. Navigate to System console > Site COnfiguration.
```
Comment (lindalumitchell) [inline data/test-cases/cloud/ip-filtering/test-1.md:46]:
```suggestion
3. Navigate to System Console > Site Configuration.
```

File: data/test-cases/cloud/ip-filtering/test-1.md [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+4. Verify that `IP Filtering` option is Not visible.
+
+Scenario 2: The Feature flag is Enabled and instance is on Enterprise trial license.
+
+1. In Split.io enable `CloudIPFiltering` feature flag for all test environments.
+2. Login with admin credentials.
+3. Navigate to System console > Site COnfiguration.
```
Comment (lindalumitchell) [inline data/test-cases/cloud/ip-filtering/test-1.md:53]:
```suggestion
3. Navigate to System Console > Site Configuration.
```

File: data/test-by-folder.json [hunk 4]
```diff
@@ -2077,6 +2077,15 @@
...
+      "IP Filtering | Add IP Filtering Rule in System console",
```
Comment (lindalumitchell) [inline data/test-by-folder.json:2084]:
```suggestion
      "IP Filtering | Add IP Filtering rule in System Console",
```

File: data/test-cases/cloud/ip-filtering/test-1.md [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+last_updated: null
+case_hashed: null
+steps_hashed: null
+---
+
+**Step 1**
+
+Test prerequisites:
+
+- Create a workspace and leave it in Enterprise Trial license from https://portal.test.cloud.mattermost.com/
```
Comment (lindalumitchell) [inline data/test-cases/cloud/ip-filtering/test-1.md:40]:
```suggestion
- Create a workspace and leave it in Enterprise trial license from https://portal.test.cloud.mattermost.com/
```

File: data/test-cases/cloud/ip-filtering/test-1.md [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+2. Login with admin credentials.
+3. Navigate to System console > Site COnfiguration.
+4. Verify that `IP Filtering` option is visible.
+
+Scenario 3: The Feature flag is Enabled and instance is on Enterprise trial license.
+
+1. In Split.io enable `CloudIPFiltering` feature flag for all test environments.
+2. Login with admin credentials.
+3. Upgrade the instance to `Cloud professional` license.
+4. Navigate to System console > Site COnfiguration.
```
Comment (lindalumitchell) [inline data/test-cases/cloud/ip-filtering/test-1.md:61]:
```suggestion
4. Navigate to System Console > Site Configuration.
```

File: data/test-cases/cloud/ip-filtering/test-1.md [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+1. In Split.io enable `CloudIPFiltering` feature flag for all test environments.
+2. Login with admin credentials.
+3. Navigate to System console > Site COnfiguration.
+4. Verify that `IP Filtering` option is visible.
+
+Scenario 3: The Feature flag is Enabled and instance is on Enterprise trial license.
+
+1. In Split.io enable `CloudIPFiltering` feature flag for all test environments.
+2. Login with admin credentials.
+3. Upgrade the instance to `Cloud professional` license.
```
Comment (lindalumitchell) [inline data/test-cases/cloud/ip-filtering/test-1.md:60]:
```suggestion
3. Upgrade the instance to `Cloud Professional` license.
```

File: data/test-cases/cloud/ip-filtering/test-2.md [hunk 8]
```diff
@@ -0,0 +1,53 @@
+---
+# (Required) Ensure all values are filled up
+name: "IP Filtering | Add IP Filtering Rule in System console"
```
Comment (lindalumitchell) [inline data/test-cases/cloud/ip-filtering/test-2.md:3]:
```suggestion
name: "IP Filtering | Add IP Filtering rule in System Console"
```

File: data/test-cases/cloud/ip-filtering/test-2.md [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+last_updated: null
+case_hashed: null
+steps_hashed: null
+---
+
+**Step 1**
+
+Test prerequisites:
+
+- Create a workspace and leave it in Enterprise Trial license from https://portal.test.cloud.mattermost.com/
```
Comment (lindalumitchell) [inline data/test-cases/cloud/ip-filtering/test-2.md:40]:
```suggestion
- Create a workspace and leave it in Enterprise trial license from https://portal.test.cloud.mattermost.com/
```

File: data/test-cases/cloud/ip-filtering/test-3.md [hunk 10]
```diff
@@ -0,0 +1,56 @@
...
+last_updated: null
+case_hashed: null
+steps_hashed: null
+---
+
+**Step 1**
+
+Test prerequisites:
+
+- Create a workspace and leave it in Enterprise Trial license from https://portal.test.cloud.mattermost.com/
```
Comment (lindalumitchell) [inline data/test-cases/cloud/ip-filtering/test-3.md:40]:
```suggestion
- Create a workspace and leave it in Enterprise trial license from https://portal.test.cloud.mattermost.com/
```

File: data/test-cases/cloud/ip-filtering/test-3.md [hunk 11]
```diff
@@ -0,0 +1,56 @@
...
+1. Login with admin credentials.
+2. Navigate to System Console > Site Configuration > IP Filtering.
+3. Toggle to enable the `Enable IP Filtering` settings.
+4. Click on `Add rule`.
+5. Enter `Name of the rule`.
+6. Enter the `IP address` other than the current IP address.
+7. Click `Save Button` on the Add IP Filter modal.
+8. Click `Save Button` on the IP Filter feature page.
+9. Click on `Yes, apply changes` to review and confirm the IP Filtering rule added.
+10. Verify the workspace is not accessible. (we should be seeing 403 error).
```
Comment (lindalumitchell) [inline data/test-cases/cloud/ip-filtering/test-3.md:52]:
```suggestion
10. Verify the workspace is not accessible (should see 403 error).
```

File: data/test-cases/cloud/ip-filtering/test-3.md [hunk 12]
```diff
@@ -0,0 +1,56 @@
...
+11. Log into portal with the workspace admin credentials
+12. Navigate to IP filtering section.
```
Comment (lindalumitchell) [inline data/test-cases/cloud/ip-filtering/test-3.md:54]:
```suggestion
12. Navigate to IP Filtering section.
```

File: data/test-cases/cloud/ip-filtering/test-3.md [hunk 13]
```diff
@@ -0,0 +1,56 @@
...
+13. Verify the `Disable IP Filtering` is enabled and Click on it.
```
Comment (lindalumitchell) [inline data/test-cases/cloud/ip-filtering/test-3.md:55]:
```suggestion
13. Verify the `Disable IP Filtering` is enabled and click on it.
```

File: data/test-cases/cloud/ip-filtering/test-4.md [hunk 14]
```diff
@@ -0,0 +1,71 @@
...
+last_updated: null
+case_hashed: null
+steps_hashed: null
+---
+
+**Step 1**
+
+Test prerequisites:
+
+- Create a workspace and leave it in Enterprise Trial license from https://portal.test.cloud.mattermost.com/
```
Comment (lindalumitchell) [inline data/test-cases/cloud/ip-filtering/test-4.md:40]:
```suggestion
- Create a workspace and leave it in Enterprise trial license from https://portal.test.cloud.mattermost.com/
```

File: data/test-cases/cloud/ip-filtering/test-4.md [hunk 15]
```diff
@@ -0,0 +1,71 @@
...
+2. Navigate to System Console > Site Configuration > IP Filtering.
+3. Toggle to enable the `Enable IP Filtering` settings.
+4. Click on `Add rule`.
+5. Enter `Name of the rule`.
+6. Enter the current `IP address`.
+7. Click `Save Button` on the Add IP Filter modal.
+8. Click `Save Button` on the IP Filter feature page.
+9. Click on `Yes, apply changes` to review and confirm the IP Filtering rule added.
+10. Verify the system admin received a email notification with subject `Changes to Your Workspace's IP Filters`.
+11. Verify the `Review changes` button in email take user to IP filtering section.
```
Comment (lindalumitchell) [inline data/test-cases/cloud/ip-filtering/test-4.md:55]:
```suggestion
11. Verify the `Review changes` button in email takes user to IP Filtering section.
```

File: data/test-cases/cloud/ip-filtering/test-4.md [hunk 16]
```diff
@@ -0,0 +1,71 @@
...
+
+Scenario: Edit Rule
+
+13. Edit the rule and alter the IP address before saving it.
+14. Verify the system admin received a email notification with subject `Changes to Your Workspace's IP Filters`
+
+Scenario: Delete Rule
+
+15. Hoven on the Rule and click on trash icon
```
Comment (lindalumitchell) [inline data/test-cases/cloud/ip-filtering/test-4.md:64]:
```suggestion
15. Hover over the rule and click on trash icon
```

File: data/test-cases/cloud/ip-filtering/test-4.md [hunk 17]
```diff
@@ -0,0 +1,71 @@
...
+16. Confirm the rule deletion.
+17. Verify the system admin received a email notification with subject `Changes to Your Workspace's IP Filters`
+
+Scenario: Enable or disable Rule
+
+18. Click on toggle button of the feature.
+19. Verify the system admin received a email notification with subject `Changes to Your Workspace's IP Filters`
```
Comment (lindalumitchell) [inline data/test-cases/cloud/ip-filtering/test-4.md:71]:
```suggestion
19. Verify the system admin received email notification with subject `Changes to Your Workspace's IP Filters`
```

**Review by lindalumitchell — APPROVED** [review]
Fantastic, thanks Yasser! A bunch of nitpicky suggestions for consistent capitalization but substance looks good. 👍

**Pull request merged**